contribute.md

```diff
@@ -3,104 +3,345 @@ layout: contribute
+
+These types of tasks generally require a familiarity with Linux, Ruby, Ruby on Rails, Javascript and React development. If you are still new to Rails, community members can help you if you get stuck with something or have any other questions.
+
+To get you started, we've prepared a short video demonstrating the process of both documented options.
+
+TODO: Put the youtube lzaps video here.
+
+#### Option A: Preconfigured virtual machine
 
 You will need to download a copy of the current development code. [The
 official code repository](https://github.com/theforeman) is located
 on GitHub. All changes are made in the `develop` branch.
 
-There is a tool called [Forklift](https://github.com/theforeman/forklift) that can be used to automate deploying Foreman development environments.
+There is a tool called [Forklift](https://github.com/theforeman/forklift) that can be used to automate deploying Foreman development environments. This will work on Linux, MacOS or even Windows and the only required tools are Vagrant (virtual machine manager and image deployment tool), Ansible (configuration management tool) and a host with virtualization capabilities.
+
+There are couple of installation profiles available, namely:
+
+* centos7-katello-devel: Git clones of foreman, remote execution and katello plugins and all required backend systems running on CentOS 7.
+* centos7-luna-devel: Git clones of foreman and all plugins which are shipped as Red Hat Satellite product running on CentOS 7.
+* centos7-foreman-nightly: Installation of nightly build of Foreman from RPM packages running on CentOS 7. This can be used for evaluation or pilot installation.
+
+Follow instructions from the Forklift github repository README, in short it is as easy as:
+
+```
+git clone https://github.com/theforeman/forklift.git
+cd forklift
+vagrant up --provision centos7-katello-devel
+vagrant ssh centos7-katello-devel
+```
+
+The last command connects as user `vagrant` with sudo permissions, then start foreman as usual:
+
+```
+vagrant# cd foreman/
+vagrant# bundle install
+vagrant# npm install
+vagrant# bundle exec foreman start
+```
+
+Connect to the VM's IP address on port 5000 to sign in as admin with password changeme. See instructions below on how to change admin's password if you forget it.
+
+#### Option B: Setting up from git manually
+
+This option is more suitable when planning working on Foreman core, the main Ruby on Rails application, core user interface components or pages or pre-configured templates which are shipped with Foreman. It is also possible to clone and enable plugin repositories, however we do not recommend to manually install Katello plugin which require number of backend systems to be installed and configured.
+
+First of all, clone our upstream repository:
+
+```
+git clone https://github.com/theforeman/foreman
```

Comment:
Just a thought: does it make sense to already name it upstream instead of origin?
```suggestion
git clone https://github.com/theforeman/foreman -o upstream
```

Then you can later add your own fork using `git remote add` without having to clone it again.

Reply:
I actually keep upstream as "origin" and my fork as "lzap". This works more nicely with quick checkouts where you want to really check out something, build it, test it but you don't contribute. I will keep it as-is if you don't mind.